daffodil-lib/src/main/scala/org/apache/daffodil/xml/DaffodilConstructingLoader.scala

```diff
@@ -288,10 +288,77 @@ class DaffodilConstructingLoader private[xml] (uri: URI,
     text(pos, s)
   }
 
+  /**
+   * Same CRLF/CR => LF processing as text gets.
+   */
   override def comment(pos: Int, s: String): Comment = {
     Comment(text(pos, s).text)
   }
 
+  /**
+   * Same CRLF/CR => LF processing as text gets.
+   */
+  override def procInstr(pos: Int, target: String, txt: String) =
+    ProcInstr(target, text(pos, txt).text)
+
+  /**
+   * Override of document to make it tolerant of the start of the file
+   * being whitespace instead of a "<" character
+   */
+  override def document(): Document = {
+
+    if ('<' == ch) super.document()
+    else {
+      // File did not start with "<".
+      // We want to tolerate whitespace and comments prior to the
+      // document element.
+
+      doc = new Document()
+
+      nextch()
+      var children: NodeSeq = null
```

Comment:
Yeah, I'm thinking I should just rewrite this method entirely instead of this super.document() thing, and then I can submit the rewritten one as the potential correction.